usr/sbin/p11sak/p11sak.c

```diff
@@ -912,6 +960,73 @@ static void set_bool_attr_from_string(CK_ATTRIBUTE *attr, char attr_char)
     }
 }
 
+/**
+ * @brief Set the custom attrs from string object
+ * 
+ * @param attr The attribute array
+ * @param attr_char The string from the user
+ * @param count Index of current attribute in the attribute array
+ */
+static CK_RV set_custom_attrs_from_string(char *attr_char, CK_ATTRIBUTE *attr, CK_ULONG *count) 
+{
+    CK_RV rc = CKR_OK;
+    int i = 0, j = 0;
+    char substr[10];
+
+    // iterate over the single custom attributes
+    while (i < (int)strlen(attr_char))
+    {
+        int radix = 16;
+        unsigned long id = 0;
+
+        // parse attribute id from string
+        for (j = 0; attr_char[i] != '='; i++) {
+            if (j == 1 && attr_char[i] != 'x' && attr_char[i] != 'X') 
+                radix = 10;
+            substr[j] = attr_char[i];
+            j++;  
+        }
+        id = strtoul(substr, NULL, radix);
+        i++;
+
+        // set attribute in array
+        attr[*count].type = id;
+        if (strncasecmp((attr_char + i), "false", 5) == 0) {
+            attr[*count].pValue = &ckb_false;
+            i += 5;
+            attr[*count].ulValueLen = sizeof(CK_BBOOL);
+        } else if (strncasecmp((attr_char + i), "true", 4) == 0) {
+            attr[*count].pValue = &ckb_true;
+            i += 4;
+            attr[*count].ulValueLen = sizeof(CK_BBOOL);
+        } else {
+            // if not boolean, a CK_ULONG value is assumed
+            radix = 16;
+            id = 0;
+            for (j = 0; attr_char[(i + j)] != ',' && attr_char[(i + j)] != '\0'; j++) {
+                if (j == 1 && attr_char[i + j] != 'x' && attr_char[i + j] != 'X')
+                    radix = 10;
```

Comment:
Set `radix = 0` and you can get rid of this loop.

Reply:
Done